src/Microsoft.OData.Core/UrlValidation/ODataUrlValidationExtensions.cs

```diff
@@ -0,0 +1,61 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataUrlValidationExtension.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using Microsoft.OData.Edm;
+using Microsoft.OData.UriParser.Validation.ValidationEngine;
+
+namespace Microsoft.OData.UriParser.Validation
+{
+    /// <summary>
+    /// Extension methods to add validation methods to an ODataUri
+    /// </summary>
+    public static class ODataUrlValidationExtensions
+    {
+        /// <summary>
+        /// Validate the OData Url for a given model using a specified set of rules.
+        /// </summary>
+        /// <param name="uri">The <see cref="Uri"/> to validate.</param>
+        /// <param name="model">The model to validate the OData Uri against.</param>
+        /// <param name="rules">The set of rules to use in validating the OData Uri.</param>
+        /// <param name="validationMessages">The collection of validation messages found during validation.</param>
+        /// <returns>True if validation messages are discovered during validation, otherwise false.</returns>
+        public static bool ValidateODataUrl(this Uri uri, IEdmModel model, ODataUrlValidationRuleSet rules, out IEnumerable<ODataUrlValidationMessage> validationMessages)

```

Comment:
ODataUriValidationMessage or ODataUriError ?

Reply:
Messages -- may be errors or warnings or information messages.

---
In reply to: [434869883](https://github.com/OData/odata.net/pull/1769#discussion_r434869883) [](ancestors = 434869883)